server/jwt_test.go

```diff
@@ -1691,6 +1691,160 @@ func TestAccountURLResolverNoFetchOnReload(t *testing.T) {
 	}
 }
 
+func TestAccountURLResolverFetchFailureInCluster(t *testing.T) {
+	const subj = "subscription.loss.test"
+	// Create Operator
+	op, _ := nkeys.CreateOperator()
+	opub, _ := op.PublicKey()
+	oc := jwt.NewOperatorClaims(opub)
+	oc.Subject = opub
+	ojwt, err := oc.Encode(op)
+	if err != nil {
+		t.Fatalf("Error generating operator JWT: %v", err)
+	}
+	// Create Account
+	akp, _ := nkeys.CreateAccount()
+	apub, _ := akp.PublicKey()
+	nac := jwt.NewAccountClaims(apub)
+	ajwt, err := nac.Encode(op)
+	if err != nil {
+		t.Fatalf("Error generating account JWT: %v", err)
+	}
+	// Create User
+	nkp, _ := nkeys.CreateUser()
+	uSeed, _ := nkp.Seed()
+	upub, _ := nkp.PublicKey()
+	nuc := newJWTTestUserClaims()
+	nuc.Subject = upub
+	uJwt, err := nuc.Encode(akp)
+	if err != nil {
+		t.Fatalf("Error generating user JWT: %v", err)
+	}
+	creds := genCredsFile(t, uJwt, uSeed)
+	defer os.Remove(creds)
+	// Simulate an account server that drops the first request to /B/acc
+	chanA := make(chan string, 4)
+	defer close(chanA)
+	chanB := make(chan string, 4)
+	defer close(chanB)
+	bCnt := int32(0)
+	errCnt := int32(0) // TODO if this is set to 1, this test will fail
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.URL.Path == "/A/" { // Server A startup
+			w.Write([]byte(ajwt))
+			chanA <- "/A/"
+		} else if r.URL.Path == "/B/" { // Server B startup
+			w.Write([]byte(ajwt))
+			chanB <- "/B/"
+		} else if r.URL.Path == "/A/"+apub { // First Client connecting to Server A
+			w.Write([]byte(ajwt))
+			chanA <- "/A/"
+		} else if r.URL.Path == "/B/"+apub { // this will be observed twice
```

Comment:
If you don't mind, could you move the comment to the inside of the "if"? I did not see that at first so other line comment looked weird.

Reply:
done